src/dataset.py

```diff
@@ -98,43 +69,33 @@ def _augment_with_salary(record: Dict[str, str]) -> Dict[str, str]:
     return context
 
 
-def badge_lookup(badge: str) -> str:
-    try:
-        r = BADGE_DATASET.get(badge)
-        if not r:
-            html = "<p><b>No officer found for this badge number</b></p>"
-        else:
-            context = _augment_with_salary(r)
-            html = render_template("officer.j2", **context)
-
-    except Exception as err:
-        print(f"Error: {err}")
-        html = f"<p><b>Error:</b> {err}"
-
-    print("Final HTML:\n{}".format(html))
-    return html
-
-
-def _sort_names(record: _DataDict):
-    for r in sorted(record.values(), key=lambda x: x[C.FIRST]):
+def _sort_names(record: List[Tuple]) -> Tuple:
+    for r in sorted(record, key=lambda x: x[3]):
         yield r
 
 
-def name_lookup(name: str) -> str:
-    try:
-        records = NAME_DATASET.get(name)
-        if not records:
-            html = "<p><b>No officer found for this name</b></p>"
-        else:
-            htmls = []
-            for r in _sort_names(records):
-                context = _augment_with_salary(r)
-                htmls.append(render_template("officer.j2", **context))
-            html = "\n<br/>\n".join(htmls)
-
-    except Exception as err:
-        print(f"Error: {err}")
-        html = f"<p><b>Error:</b> {err}"
-
-    print("Final HTML:\n{}".format(html))
-    return html
+def name_lookup(last_name: Any, first_name: Any, badge: Any) -> str:
+    if last_name or first_name or badge:
+        try:
+            records = sql_curs.execute(
+                "SELECT * FROM officers WHERE Last = ? OR First = ? OR Serial = ?",
```

Comment:
I would suggest using `LIKE` rather than a hard match - this will allow users to make close approximation searches (most DBs have `ILIKE` for case insensitive like, but it seems `LIKE` is [case insensitive in sqlite](https://www.sqlitetutorial.net/sqlite-like/))
According to [this SO question](https://stackoverflow.com/a/3105370/3277713), you'll have to add the wildcards (`%`) to the string prior to executing the query. We won't want to do this to `None` values that are supplied to the function because I think that will make it match all records lol.

Reply:
Yeah I need to play around with the SQL query options more. I'm going to be updating this to be a little more dynamic as well, and doing something similar to:

```
>>> sql_query = "SELECT * FROM officers WHERE Last = ?"
>>> sql_curs.execute(sql_query, ("Aagard",),).fetchall()
[('5011', 'Lori', 'K.', 'Aagard', 'Police Lieutenant', 'South Pct Ops')]
>>> sql_query += " OR First = ?"
>>> sql_query
'SELECT * FROM officers WHERE Last = ? OR First = ?'
>>> sql_curs.execute(sql_query, ("Aagard","Lori",),).fetchall()
[('5011', 'Lori', 'K.', 'Aagard', 'Police Lieutenant', 'South Pct Ops'), ('5819', 'Lori', 'A', 'Nelsen-Allen', 'Admin Specialist I', 'Dv - Domestic Violence Unit - Admi~')]
```

I didn't realize you could do this at first but it makes a lot more sense and also will simplify the query param gets above so that it doesn't matter if they're `str(None)` or just `None`.

Reply:
Yea it's a very common pattern to have the base SQL statement and then build up the `WHERE` clauses as needed with different parameters!